brigade/core/exceptions.py

```diff
@@ -51,3 +51,24 @@ def __str__(self):
                 text += "**** {}\n".format(sub_r.name)
                 text += "{}\n".format(sub_r)
         return text
+
+class BrigadeSubTaskError(Exception):
+    """
+    Raised by brigade when a sub task managed by :meth:`brigade.core.Task.run`
+    has failed
+
+    Arguments:
+        task (str): Name of the sub task that failed
```

Comment:
I think task should be a reference the task itself, not just the name. I think we should also have a reference to the result. So you'd basically call this like (see comment below)